core/src/main/java/com/sequenceiq/cloudbreak/reactor/handler/cluster/datalakemetrics/datasizes/GetDatalakeDataSizesHandler.java

```diff
@@ -78,16 +82,25 @@ protected Selectable doAccept(HandlerEvent<GetDatalakeDataSizesRequest> event) {
         Long stackId = request.getResourceId();
         StackDto stackDto = stackDtoService.getById(stackId);
         GatewayConfig primaryGatewayConfig = gatewayConfigService.getPrimaryGatewayConfig(stackDto);
-
+        String gatewayHost = primaryGatewayConfig.getHostname();
+        Set<String> serviceHosts = getServiceHosts(stackDto, gatewayHost);
+        if (serviceHosts.isEmpty()) {
+            serviceHosts = Set.of(gatewayHost);
+        }
```

Comment:
We don't need this check since the return of getServicesHosts is `return coreHosts.isEmpty() ? Set.of(primaryGatewayHost) : coreHosts;` it is never  true.

Reply:
nice catch, thanks. Updated with `assert`